### PR TITLE
Add image dump in PNG

### PR DESCRIPTION
**Pull request by dj2**
This change adds support to dump image to PNG format, it relies on the
third party 'lodepng' library (zlib license).

This CL is based on Pull #301 by @hevrard.

### CODE REVIEW

**Comment by dj2** [comment]
@hevrard I think this should work ....

File: samples/amber.cc [hunk 1]
```diff
@@ -325,10 +326,18 @@ int main(int argc, const char** argv) {
 
     if (!options.image_filename.empty()) {
       std::string image;
+      bool usePNG = options.image_filename.compare(
+          options.image_filename.length() - 5, options.image_filename.length(),
```
Comment (dneto0) [inline samples/amber.cc]: What if the string is very short, e.g. length is 2, then 2-5 is large negative that is then converted to unsigned....  out of bounds?

File: samples/png.cc [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wweak-vtables"
+#include "third_party/lodepng/lodepng.h"
+#pragma clang diagnostic pop
+
+namespace png {
+
+namespace {
+
+unsigned char byte0(uint32_t word) {
```
Comment (dneto0) [inline samples/png.cc]: Normally I'd ask for uint8_t here. But lodepng wants a vector of unsigned char.  So I'll let this pass.

File: samples/png.cc [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  }
+
+  std::vector<unsigned char> png;
+  unsigned error = lodepng::encode(png, data, width, height);
+  if (error) {
+    return std::make_pair(amber::Result("lodepng::encode() returned non-zero"),
+                          nullptr);
+  }
+
+  std::string image;
```
Comment (dneto0) [inline samples/png.cc]: Would recommend this instead:
std::string image(png.data(), png.size())

File: samples/amber.cc (hunk 1 above)
Comment (dj2) [reply samples/amber.cc]: done

File: samples/png.cc (hunk 2 above)
Comment (dj2) [reply samples/png.cc]: ack

File: samples/png.cc (hunk 3 above)
Comment (dj2) [reply samples/png.cc]: std::string(png.begin(), png.end()) works

**Pull request merged**